### PR TITLE
Feat/ams form additions

### PR DESCRIPTION
**Pull request by rhofvendahl**
Add additional fields & schemas to improve `USDASpecialtyCrops237AForm`, in response to stakeholder feedback.

### CODE REVIEW

File: docs/openapi/components/schemas/common/AgricultureProduct.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ properties:
     type: string
     $linkedData:
       term: plu
-      '@id': https://w3id.org/traceability#plu
+      '@id': https://schema.org/identifier
```
Comment (OR13) [inline docs/openapi/components/schemas/common/AgricultureProduct.yml:43]: I searched a bit for a better term and did not find one. 

https://www.pma.com/content/articles/2014/05/plu-codes <-- 500 error for me.

Comment (OR13) [reply docs/openapi/components/schemas/common/AgricultureProduct.yml:43]: maybe better: https://www.ifpsglobal.com/PLU-Codes

File: docs/openapi/components/schemas/common/AgricultureProduct.yml [hunk 2]
```diff
@@ -120,6 +120,27 @@ properties:
     $linkedData:
       term: productImageHash
       '@id': https://w3id.org/traceability#productImageHash
+  variety:
+    title: Variety
+    description: "Additional details regarding the product. For example, in this case of potatoes this might include: russet, white, yellow flesh, etc."
+    type: string
+    $linkedData:
+      term: variety
+      '@id': https://www.gs1.org/voc/additionalProductDescription
```
Comment (OR13) [inline docs/openapi/components/schemas/common/AgricultureProduct.yml:129]: good choice

File: docs/openapi/components/schemas/common/AgricultureProduct.yml [hunk 3]
```diff
@@ -120,6 +120,27 @@ properties:
...
+  commodityDesignation:
+    title: Commodity Designation
+    description: "Additional product subcategories such as: frozen, fresh, sliced, canned, dried, dehydrated, etc."
+    type: string
+    $linkedData:
+      term: commodityDesignation
+      '@id': https://www.gs1.org/voc/additionalProductDescription
```
Comment (OR13) [inline docs/openapi/components/schemas/common/AgricultureProduct.yml:136]: repeating this here will cause problems in the graph format.

File: docs/openapi/components/schemas/common/AgricultureProduct.yml [hunk 4]
```diff
@@ -120,6 +120,27 @@ properties:
...
+  packType:
+    title: Pack Type
+    description: "Details like: volume filled, tray pack, 18/lb consumer units, 10/5 pound bags, etc."
+    type: string
+    $linkedData:
+      term: packType
+      '@id': AAAA
```
Comment (OR13) [inline docs/openapi/components/schemas/common/AgricultureProduct.yml:143]: this is invalid

File: docs/openapi/components/schemas/common/ExtendedProduct.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    $linkedData:
+      term: supplier
+      '@id': https://www.gs1.org/voc/manufacturer
+  htsCode:
+    title: HTS Code
+    description: The Harmonized Tariff Schedule (HTS) code.
+    type: string
+    $linkedData:
+      term: htsCode
+      '@id': https://vocabulary.uncefact.org/applicableRegulatoryProcedure
```
Comment (OR13) [inline docs/openapi/components/schemas/common/ExtendedProduct.yml:39]: seems like there is probably a better term for this, but I did not find one easily

**Review by OR13 — CHANGES_REQUESTED** [review]
only blocking feedback is the invalid IRI.

File: docs/openapi/components/schemas/common/AgricultureProduct.yml (hunk 4 above)
Comment (rhofvendahl) [reply docs/openapi/components/schemas/common/AgricultureProduct.yml:143]: Shoot, left this in by mistake... thanks for catching it

File: docs/openapi/components/schemas/common/ExtendedProduct.yml (hunk 5 above)
Comment (rhofvendahl) [reply docs/openapi/components/schemas/common/ExtendedProduct.yml:39]: I can switch this to `applicableTax` ("A tax applicable to this header, line, subordinate line or payment trade settlement or cross-border regulatory procedure."), to be a bit more specific

**Review by mkhraisha — APPROVED** [review]

**Comment by rhofvendahl** [comment]
I'm closing this for now, to be re-opened when I've finished the refactor suggested on last week's call.

**Comment by rhofvendahl** [comment]
I've refactored ExtendedProduct into AgProduct and Product, which should be a lot cleaner.

File: docs/openapi/components/schemas/common/AgricultureProduct.yml [hunk 6]
```diff
@@ -76,7 +69,51 @@ properties:
     type: string
     $linkedData:
       term: labelImageHash
-      '@id': https://schema.org/sha256
+      '@id': https://w3id.org/traceability#labelImageHash
+  name:
+    title: Name
+    description: Name of the ag. product.
```
Comment (TallTed) [inline docs/openapi/components/schemas/common/AgricultureProduct.yml:75]:
```suggestion
    description: Name of the agricultural product.
```

**Review by TallTed — CHANGES_REQUESTED** [review]
tweaks

**Review by nissimsan — APPROVED** [review]

File: docs/openapi/components/schemas/common/AgricultureProduct.yml [hunk 7]
```diff
@@ -76,7 +69,51 @@ properties:
     type: string
     $linkedData:
       term: labelImageHash
-      '@id': https://schema.org/sha256
+      '@id': https://w3id.org/traceability#labelImageHash
```
Comment (OR13) [inline docs/openapi/components/schemas/common/AgricultureProduct.yml:72]: too bad we don't have a better term def for this

**Review by OR13 — APPROVED** [review]

**Pull request merged**